x/concentrated-liquidity/genesis.go

```diff
@@ -244,3 +261,63 @@ func (k Keeper) initOrUpdateAccumPosition(ctx sdk.Context, accumumulator accum.A
 
 	osmoutils.MustSet(ctx.KVStore(k.storeKey), accum.FormatPositionPrefixKey(accumumulator.GetName(), index), &position)
 }
+
+// GetAllInitializedTicksForPoolWithoutPoolId returns all ticks in FullTick struct, without the pool id.
+func (k Keeper) GetAllInitializedTicksForPoolWithoutPoolId(ctx sdk.Context, poolId uint64) ([]genesis.FullTick, error) {
+	return osmoutils.GatherValuesFromStorePrefixWithKeyParser(ctx.KVStore(k.storeKey), types.KeyTickPrefixByPoolId(poolId), ParseFullTickFromBytesWithoutPoolId)
+}
+
+// ParseFullTickFromBytesWithoutPoolId takes key and value byte slices and attempts to parse
+// them into a FullTick struct. If the key or value is not valid, an appropriate
+// error is returned. The function expects the key to have three components
+// 1. The tick prefix (1 byte)
+// 2. The pool id (8 bytes)
+// 3. The tick index (1 byte for sign + 8 bytes for unsigned integer)
+//
+// The function returns a FullTick struct containing the pool id, tick index, and
+// tick information.
```

Comment:
I thought pool id is not returned. Could you update the spec, please?

Also, why is it important to have "WithoutPoolId` in the name?

Reply:
I think I have used such name since the change I was making no longer returned pool Id with it. Coming back to it, I think it's fine to remove it, changed to ParseFullTickFromBytes

Reply:
Spec also updated